src/core/ext/filters/client_channel/lb_policy/subchannel_list.cc

```diff
@@ -0,0 +1,286 @@
+      // Subchannel could not be created.
+      if (GRPC_TRACER_ON(*tracer)) {
+        char *address_uri =
+            grpc_sockaddr_to_uri(&addresses->addresses[i].address);
+        gpr_log(GPR_DEBUG,
+                "[%s %p] could not create subchannel for address uri %s, "
+                "ignoring",
+                tracer->name, subchannel_list->policy, address_uri);
+        gpr_free(address_uri);
+      }
+      continue;
+    }
+    grpc_error *error;
+    // Get the connectivity state of the subchannel. Already existing ones may
+    // be in a state other than INIT.
+    const grpc_connectivity_state subchannel_connectivity_state =
+        grpc_subchannel_check_connectivity(subchannel, &error);
+    if (error != GRPC_ERROR_NONE) {
+      // The subchannel is in error (e.g. shutting down). Ignore it.
+      if (GRPC_TRACER_ON(*tracer)) {
+        char *address_uri =
+            grpc_sockaddr_to_uri(&addresses->addresses[i].address);
+        gpr_log(GPR_DEBUG,
+                "[%s %p] subchannel for address uri %s shutting down, ignoring",
+                tracer->name, subchannel_list->policy, address_uri);
+        gpr_free(address_uri);
+      }
+      GRPC_SUBCHANNEL_UNREF(exec_ctx, subchannel, "new_sc_connectivity_error");
+      GRPC_ERROR_UNREF(error);
+      continue;
+    }
+    if (GRPC_TRACER_ON(*tracer)) {
+      char *address_uri =
+          grpc_sockaddr_to_uri(&addresses->addresses[i].address);
+      gpr_log(GPR_DEBUG, "[%s %p] subchannel list %p index %" PRIdPTR
+                         ": Created subchannel %p for address uri %s; "
+                         "initial connectivity state: %s",
+              tracer->name, p, subchannel_list, subchannel_index, subchannel,
+              address_uri,
+              grpc_connectivity_state_name(subchannel_connectivity_state));
+      gpr_free(address_uri);
+    }
+    grpc_lb_subchannel_data *sd =
+        &subchannel_list->subchannels[subchannel_index++];
+    sd->subchannel_list = subchannel_list;
+    sd->subchannel = subchannel;
+    GRPC_CLOSURE_INIT(&sd->connectivity_changed_closure,
+                      connectivity_changed_cb, sd,
+                      grpc_combiner_scheduler(args->combiner));
+    /* use some sentinel value outside of the range of
+     * grpc_connectivity_state to signal an undefined previous state. We
+     * won't be referring to this value again and it'll be overwritten after
+     * the first call to rr_connectivity_changed_locked */
+    sd->prev_connectivity_state = GRPC_CHANNEL_INIT;
+    sd->curr_connectivity_state = subchannel_connectivity_state;
+    sd->user_data_vtable = addresses->user_data_vtable;
+    if (sd->user_data_vtable != NULL) {
+      sd->user_data =
+          sd->user_data_vtable->copy(addresses->addresses[i].user_data);
+    }
+  }
+  subchannel_list->num_subchannels = subchannel_index;
+  return subchannel_list;
+}
+
+static void subchannel_list_destroy(grpc_exec_ctx *exec_ctx,
```

Comment:
There's no need for the `grpc_` prefix on a static function.  It's not visible outside of this file, so there's no namespace reason to fully qualify it.

Reply:
Good point.  This is because `grpc_lb_subchannel_data_cancel_connectivity_watch()` was originally a non-static function, exposed via the header file, until I noticed that it wasn't being used externally anywhere.  But I forgot to remove the `grpc_lb_` prefix when I made it a static function.  I've fixed that now.